solr/core/src/java/org/apache/solr/store/blob/metadata/SharedStoreResolutionUtil.java

```diff
@@ -147,42 +139,53 @@ public static SharedMetadataResolutionResult resolveMetadata(ServerSideMetadata
     
     if (local == null) {
       // The shard index data does not exist locally. All we can do is pull.  
-      // We've computed blobFilesMissingLocally and localFilesMissingOnBlob is empty as it should be.
+      // We've computed blobFilesMissingLocally. localFilesMissingOnBlob is empty as it should be.
       return new SharedMetadataResolutionResult(localFilesMissingOnBlob.values(), blobFilesMissingLocally.values(), blobFilesMissingLocally.values(), false);
     }
 
-    boolean localConflictingWithBlob = false;
+    // If trying to pull files from Blob, make sure similarly named files do not already exist outside the current commit point
+    ImmutableSet<String> allLocalFiles = local.getAllFiles();
+
+    boolean downloadToNewDir = false;
```

Comment:
Maybe pullToNewDir? I don't have a strong preference but it would be better remain
consistent and use the same name here https://github.com/apache/lucene-solr/pull/1055/files?utf8=%E2%9C%93&diff=unified&w=1#diff-29111919397c5d8b89434d02aab74127R41.

Reply:
pulling from Blob downloads files :)
Not sure I understand what the link pasted above points to.

Reply:
The link is not working for me anymore either(it was pointing to a line in the code diff).
It was pointing to SharedMetadataResolutionResult#localConflictingWithBlob.
I was suggesting to keep that consistent with your rename here.